llvm/lib/Target/NVPTX/NVPTXISelLowering.h

```diff
@@ -496,12 +496,20 @@ class NVPTXTargetLowering : public TargetLowering {
 
   bool isTruncateFree(Type *SrcTy, Type *DstTy) const override {
     // Truncating 64-bit to 32-bit is free in SASS.
-    if (!SrcTy->isIntegerTy() || !DstTy->isIntegerTy())
+    if (!(SrcTy->isIntegerTy() && DstTy->isIntegerTy()))
```

Comment:
NFC - do this as a separate patch if you must

Reply:
@justinfargnoli I think this comment has none been responded to?

Reply:
demorgan this condition (i.e. just remove this change altogether)